theano/gof/utils.py

```diff
@@ -210,6 +211,47 @@ def __str__(self):
 
         return type.__new__(cls, name, bases, dct)
 
+    _cache = {}
+
+    def __call__(cls, *args, **kwargs):
+        _set_cache = cls.__dict__.get('_set_cache', None)
+        if _set_cache:
+            key = [cls]
+            for arg in args:
+                arg = make_hashable(arg)
+                key.append(arg)
+            for k, v in sorted(kwargs.items()):
+                key.append(k)
+                v = make_hashable(v)
```

Comment:
@nouiz , @abergeron , right now we are only caching the 6 ops that had the caching implementation in master. In these 6 ops, ```make_hashable()``` is never used. I can safely remove it. Since at the start we were caching all the ops with ```__props___```, we had so many weird ```args``` and ```kwargs``` so we implemented the ```make_hashable()```.  We can comment it for now and reuse it if necessary in the future. What to you think?